tests/main/snapd-homedirs-vendored/task.yaml

```diff
@@ -13,8 +13,8 @@ environment:
     USERNAME: home-sweet-home
 
 prepare: |
-    if tests.info is-snapd-from-archive; then
-        tests.exec skip-test "No needed to test the snap pkg when it is coming from the repository" && exit 0
+    if [[ "$SNAP_REEXEC" = "0" ]]  || tests.info is-snapd-from-archive; then
```

Comment:
This scenario is not exercised as tests are run for this PR, but was specifically tested locally with SPREAD_SNAPD_DEB_FROM_REPO=false SPREAD_MODIFY_CORE_SNAP_FOR_REEXEC=0 SPREAD_TRUST_TEST_KEYS=false SPREAD_SNAP_REEXEC=0 SPREAD_CORE_CHANNEL=stable.